src/ddmd/func.d

```diff
@@ -156,6 +156,61 @@ enum FUNCFLAGinlineScanned    = 0x20;   /// function has been scanned for inline
 enum FUNCFLAGinferScope       = 0x40;   /// infer 'scope' for parameters
 
 
+/****************************************************
+ * Determine whether an 'in' contract is declared inside
+ * the given function or any of its overrides.
+ * Params:
+ *      fd = the function to search
+ * Returns:
+ *      true    found an 'in' contract
+ *      false   didn't find one
+ */
+private bool needsFrequire(FuncDeclaration fd)
+{
+    if (fd.frequire)
+        return true;
+
+    for (size_t i = 0; i < fd.foverrides.dim; i++)
+    {
+        FuncDeclaration fdv = fd.foverrides[i];
```

Comment:
```
foreach (fdv; fd.overrides)
```
should do it.

Reply:
Yep. I think it will take me a while longer to get back to using foreach.